pkg/pillar/cmd/diag/diag.go

```diff
@@ -20,24 +28,11 @@ import (
 	"github.com/lf-edge/eve/pkg/pillar/types"
 	"github.com/lf-edge/eve/pkg/pillar/zedcloud"
 	log "github.com/sirupsen/logrus"
-	"io"
-	"io/ioutil"
-	"net"
-	"net/http"
-	"os"
-	"strings"
-	"time"
 )
 
 const (
-	agentName       = "diag"
-	identityDirname = "/config"
-	selfRegFile     = identityDirname + "/self-register-failed"
```

Comment:
selfRegFile is only used by diag. So why promote it to common?

Reply:
Hmm.. Who creates this file? grep doesn't seem to indicate ?

Hmm.. Should this really . be "self-register-pending" ?? That is the file being created by device-steps.sh. 

Hmm.. looking at this file - Client just doesn't create this file. It seems to be in this infinite loop trying to self register. Should diag be fixed??

Reply:
Yes, it should be self-register-pending. Diag just looks at it for its output.

What infinite loop??